### PR TITLE
Dataloader and Noise

### PR DESCRIPTION
**Pull request by gbruno16**
# Pull Request

## Description
This PR implements the dataset and the noise generation for GenCast, as detailed in #106 .

### CODE REVIEW

**Review by peterdudfield — APPROVED** [review]

File: graph_weather/data/const.py [hunk 1]
```diff
@@ -5638,3 +5640,475 @@
     "WEASD.surface": -12.248001098632812,
     "WILT.surface": -0.004800006747245789,
 }
+
+
+ERA5_MEANS = {
```
Comment (peterdudfield) [inline graph_weather/data/const.py:5645]: do you mind add a comment why this is a 1-D array?

Comment (gbruno16) [reply graph_weather/data/const.py:5650]: Done!

Comment (peterdudfield) [reply graph_weather/data/const.py:5650]: Thanks, this is perfect

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
@gbruno16 feel free to merge

File: graph_weather/data/gencast_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+            axis=-1,
+        ).astype(np.float32)
+        static_input_data = np.stack([static_input_data] * 2, axis=0)
+        static_input_data = np.nan_to_num(static_input_data)
+        assert not np.isnan(static_input_data).any()
+        assert (static_input_data[0] == static_input_data[1]).all()
+
+        # Stack the time features for input
+        day_of_year = start.time.dt.dayofyear.values / 365.0
+        sin_day_of_year = (
```
Comment (Sukh-P) [inline graph_weather/data/gencast_dataloader.py:118]: Again this is a nit but I can see these transformations are used a few times and look very similar here, could be pulled out into a helper function where you just specify whether you want to use sin or cos to transform the day of the year

Comment (gbruno16) [reply graph_weather/data/gencast_dataloader.py]: You're absolutely right! I've just reorganized the functions to reuse them when possible (sin, cos, normalization, etc.)

File: graph_weather/data/IFSAnalysis_dataloader.py [hunk 3]
```diff
@@ -23,7 +27,21 @@
 
 
 class IFSAnalisysDataset(Dataset):
+    """
+    Dataset for IFSAnalysis.
+
+    Args:
+            filepath (str): path of the dataset.
+            features (list): list of features.
+            start_year (int, optional): initial year. Defaults to 2016.
```
[the remark this answers is not in the document]
Comment (gbruno16) [reply graph_weather/data/IFSAnalysis_dataloader.py]: Done! I've removed the type hints from the docstring and added them in the __init__ function of the GenCast dataset. The 'optional' was added by my autodocstring extension, even though it's not optional, just deleted it.

**Comment by peterdudfield** [comment]
@all-contributors please add @gbruno16  for code

**Pull request merged**